tests/rkt_service_file_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/coreos/go-systemd/dbus"
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/coreos/go-systemd/util"
+)
+
+func TestServiceFile(t *testing.T) {
+	if !util.IsRunningSystemd() {
+		t.Skip("Systemd is not running on the host.")
+	}
+
+	ctx := newRktRunCtx()
+	defer ctx.cleanup()
+
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	conn, err := dbus.New()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	image, err := filepath.Abs("rkt-inspect.aci")
+	if err != nil {
+		t.Fatal(err)
+	}
+	opts := "-- --print-msg=HelloWorld --sleep=1000"
+
+	cmd := fmt.Sprintf("%s --insecure-skip-verify run -set-env=MESSAGE_LOOP=1000 %s %s", ctx.cmd(), image, opts)
+	props := []dbus.Property{
+		dbus.PropExecStart(strings.Split(cmd, " "), false),
+	}
+	target := fmt.Sprintf("rkt-testing-transient-%d.service", r.Int())
+
+	reschan := make(chan string)
+	_, err = conn.StartTransientUnit(target, "replace", props, reschan)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	job := <-reschan
+	if job != "done" {
+		t.Fatal("Job is not done:", job)
+	}
+
+	units, err := conn.ListUnits()
+
+	var found bool
+	for _, u := range units {
+		if u.Name == target {
+			found = true
+			if u.ActiveState != "active" {
+				t.Fatalf("Test unit %s not active: %s (target: %s)", u.Name, u.ActiveState, target)
+			}
+		}
+	}
+
+	if !found {
+		t.Fatalf("Test unit not found in list")
+	}
+
+	// Run the unit for 10 seconds. You can check the logs manually in journalctl
```

Comment:
No... at the moment, it is more a manual test and we can run `journalctl` commands to see the output of the app printed every second. 10 seconds is enough time for `rkt` to start the app. Ideally, the test would run `journalctl` commands itself and check the output.

We will need more functional tests based on a systemd service file to check the app isolators (#811) & pods isolators (#293) when they get implemented. It will need refactoring.